test/functional/tool_wallet.py

```diff
@@ -61,10 +61,122 @@ def wallet_timestamp(self):
+        self.log_wallet_timestamp_comparison(timestamp_before, timestamp_after)
+
+        # Restore permissions
+        self.log.debug('Setting wallet file permissions back to 600 (read/write)')
+        os.chmod(self.wallet_path, stat.S_IRUSR | stat.S_IWUSR)
+        permissions_after = self.wallet_permissions()
+        self.log.debug(f'Wallet permissions after restoring read-write: {permissions_after}')
+        assert self.is_wallet_readable(), f"Wallet file should be readable after restoring permissions for {command_name}"
+
+        # Verify wallet file unchanged
+        assert_equal(timestamp_before, timestamp_after)
+        shasum_after = self.wallet_shasum()
+        assert_equal(shasum_before, shasum_after)
+        self.log.debug(f'Wallet file shasum unchanged after {command_name}\n')
+
+    def test_tool_command_wallet_unchanged(self, command_name, tool_args, expected_output):
+        """
+        Helper method to test wallet tool commands that should not modify the wallet file.
+        Records wallet state before/after and verifies no changes occurred.
+
+        Args:
+            command_name: Name of the command being tested (for logging)
+            tool_args: List of arguments to pass to bitcoin-wallet tool
+            expected_output: Expected stdout output
+        """
+        shasum_before = self.wallet_shasum()
+        timestamp_before = self.wallet_timestamp()
+        self.log.debug(f'Wallet file timestamp before calling {command_name}: {timestamp_before}')
+
+        self.assert_tool_output(expected_output, *tool_args)
+
+        shasum_after = self.wallet_shasum()
+        timestamp_after = self.wallet_timestamp()
+        self.log.debug(f'Wallet file timestamp after calling {command_name}: {timestamp_after}')
+        self.log_wallet_timestamp_comparison(timestamp_before, timestamp_after)
+
+        # Verify wallet file unchanged
+        assert_equal(timestamp_before, timestamp_after)
+        assert_equal(shasum_before, shasum_after)
+        self.log.debug(f'Wallet file shasum unchanged after {command_name}\n')
+
+    def verify_wallet_file_unchanged(self, command_name, shasum_before, timestamp_before):
+        """
+        Helper method to verify wallet file was not modified after an operation.
+
+        Args:
+            command_name: Name of the command being tested (for logging)
+            shasum_before: Wallet checksum before operation
+            timestamp_before: Wallet timestamp before operation
+        """
+        shasum_after = self.wallet_shasum()
+        timestamp_after = self.wallet_timestamp()
+        self.log.debug(f'Wallet file timestamp after calling {command_name}: {timestamp_after}')
+        self.log_wallet_timestamp_comparison(timestamp_before, timestamp_after)
+
+        assert_equal(timestamp_before, timestamp_after)
+        assert_equal(shasum_before, shasum_after)
+        self.log.debug(f'Wallet file shasum unchanged after {command_name}\n')
```

Comment:
I think these can be further deduplicated by passing in a function to `test_tool_command_wallet_unchanged` which does some action on the wallet that should not have changed it. Then callers can define whatever they need that function to do, rather than having these separate cases for wallet tool without output, wallet tool with output, and other commands.